### PR TITLE
fix (next): updated getting started guide after v5 tests

### PR DESCRIPTION
**Pull request by cwomack**
_Issue #, if available:_

_Description of changes:_
Fixes to the Next.JS documentation for Getting Started with Amplify after `No current user` errors occurred when following steps in guide.  Everything works locally, however there is an issue with Hosting tied to https://github.com/vercel/next.js/issues/41932 and the Frontend build failing for CI/CD.  The error in the logs is the following: 

Screen Shot 2022-11-09 at 12 17 06 PM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

File: src/fragments/start/getting-started/next/api.mdx [hunk 1]
```diff
@@ -89,16 +95,18 @@ amplify push
...
-✔ Enter a description for the API key: ·
-✔ After how many days from now the API key should expire (1-365): · 7
-GraphQL schema compiled successfully.
-
-? Do you want to generate code for your newly created GraphQL API Yes
-? Choose the code generation language target javascript
-? Enter the file name pattern of graphql queries, mutations and subscriptions src/graphql/**/*.js
-? Do you want to generate/update all possible GraphQL operations - queries, mutations and subscriptions Yes
-? Enter maximum statement depth [increase from default if your schema is deeply nested] 2
+✔ Enter a description for the API key: · (Just hit enter)
```
Comment (tannerabread) [inline src/fragments/start/getting-started/next/api.mdx:98]: for all of these `(Just hit enter)` would it not be better to do `(default)` or does it not suggest a default here?

File: src/fragments/start/getting-started/next/api.mdx [hunk 2]
```diff
@@ -346,7 +360,10 @@ Submit that form to create a new post, and you'll build that page next!
 
 Statically generating pages during the build process improves performance. But, dynamically created posts still need to not `404`.
 
-To solve this, create **pages/posts/[id].js** and paste in the following content:
+To solve this, create a new folder within **/pages** named "posts".  Next, create a new file within this new **/posts** named **[id].js** to create the structure:
```
Comment (tannerabread) [inline src/fragments/start/getting-started/next/api.mdx:363]: this change seems unnecessary to me but if it slowed you down at all while you were running through the guide go ahead with it

File: src/fragments/start/getting-started/next/api.mdx [hunk 3]
```diff
@@ -359,29 +376,16 @@ import styles from '../../styles/Home.module.css';
...
-
-  return {
-    fallback: true,
-    paths
-  };
-}
-
-export async function getStaticProps({ params }) {
-  const SSR = withSSRContext();
+export async function getServerSideProps({req, params}) {
```
Comment (tannerabread) [inline src/fragments/start/getting-started/next/api.mdx:379]:
```suggestion
export async function getServerSideProps({ req, params }) {
```

File: src/fragments/start/getting-started/next/api.mdx [hunk 4]
```diff
@@ -359,29 +376,16 @@ import styles from '../../styles/Home.module.css';
...
+  const SSR = withSSRContext({req});
```
Comment (tannerabread) [inline src/fragments/start/getting-started/next/api.mdx:380]:
```suggestion
  const SSR = withSSRContext({ req });
```

File: src/fragments/start/getting-started/next/api.mdx [hunk 5]
```diff
@@ -359,29 +376,16 @@ import styles from '../../styles/Home.module.css';
...
   const { data } = await SSR.API.graphql({
     query: getPost,
     variables: {
       id: params.id
     }
   });
-
-  return {
+  console.log(data)
+  return {
```
Comment (tannerabread) [inline src/fragments/start/getting-started/next/api.mdx:388]:
```suggestion
  return {
```

**Review by tannerabread — COMMENTED** [review]
Mostly looks good to me just wanted to discuss those few changes first

File: src/fragments/start/getting-started/next/api.mdx (hunk 2 above)
Comment (cwomack) [reply src/fragments/start/getting-started/next/api.mdx:363]: That's fair feedback.  I felt it was a little verbose too.  I'll rework it to something shorter and/or just revert to original.

File: src/fragments/start/getting-started/next/api.mdx (hunk 1 above)
Comment (cwomack) [reply src/fragments/start/getting-started/next/api.mdx:98]: It doesn't necessarily spell out a default, and the docs just have that dot/apostrophe that may be hard to see.  

But (default) should work just fine and I'll make that change.

**Review by tannerabread — APPROVED** [review]
LGTM  🚀 🚀 🚀

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

**Pull request merged**